src/tiler/server.js

```diff
@@ -48,37 +46,77 @@ var Windshaft = require('windshaft'),
+        },
 
-            try {
-                if (req.params.type == 'progress') {
-                    req.params.table = 'survey_blockface';
-                    req.params.interactivity = 'id,group_id,survey_type';
-
-                    req.params.style = progressStyle;
-
-                    if ('user' in req.query) {
-                        user_id = parseInt(req.query.user, 10);
-                        if (isNaN(user_id)) {
-                            callback("Could not parse user id", null);
-                            return;
-                        }
-                        req.params.sql = userProgressSql({user_id: user_id});
-                    } else {
-                        req.params.sql = progressSql;
-                    }
-
-                } else {
-                    callback("Unrecognized request type", null);
-                    return;
+        req2interactivity: function(req) {
+            // At the time this was written we have the same interactivity
+            // columns for all requests
+            return 'id,group_id,survey_type';
+        },
+
+        req2style: function(req) {
+            // At the time this was written we have a single stylesheet
+            // for all requests
+            return styles.progress();
+        },
+
+        req2sql: function(req) {
+            /*
+            This function expects the SQL files loaded into the
+            'queries' object to have a specific naming convention. If
+            a user= query string argument is provided, this funciton
+            prepends 'user_' to the 'type' parameter to generate the
+            SQL file name. Example:
+
+            URL:  /123456/nyc_trees/foo/1/2/3.png
+            File: foo.sql
+
+            URL:  /123456/nyc_trees/foo/1/2/3.png?user=1
+            File: user_foo.sql
+
+            */
+            var queryPrefix = req.query.user ? 'user_' : '',
+                queryName = queryPrefix + req.params.type,
+                context = this.req2context(req),
+                additionalErr = '';
+            if (queries[queryName]) {
+                return queries[queryName](context);
+            } else {
+                if (queries['user_' + queryName]) {
+                    additionalErr = 'There is a query defined for user_' + queryName +
+                        '. Did you forget the user= query string argument?';
                 }
+                throw 'No query defined for ' + queryName + '. ' + additionalErr;
+            }
+        },
+
+        req2params: function(req, callback) {
+            try {
+                // Windshaft needs a table name, even if you are providing
+                // a custom sql statement.
+                req.params.table = 'survey_blockface';
+                req.params.sql = this.req2sql(req);
+                req.params.style = this.req2style(req);
+                req.params.interactivity = this.req2interactivity(req);
+                callback(null, req);
```

Comment:
They were initially all standalone functions, but some of them are not pure and depend on elements of the config object. I tried leaving the pure functions standalone, but then similar functions were in separate places, which made it harder to follow what was going on.